mercury/views/views.py

```diff
@@ -9,3 +11,22 @@ def get(self, request, **kwargs):
 
 class AboutPageView(TemplateView):
     template_name = "about.html"
+
+
+class EventAccess(TemplateView):
+    def post(self, request, *args, **kwargs):
+        event_code = request.POST.get("eventcode")
+        event_access_object = EventCodeAccess.objects.latest("id")
+        event_code_actual = event_access_object.event_code
+        if event_code == event_code_actual:
+            return HttpResponseRedirect("index")
+        else:
+            return HttpResponse("Invalid Event Code")
+
+    def get(self, request, **kwargs):
+        event_access_object = EventCodeAccess.objects.latest("id")
```

Comment:
I'm not sure I understand the behavior here. If someone tries to GET the login page, it checks the latest event code, and brings you to the "homepage" (index.html) if it is _not_ enabled? Can describe more the intention of the functionality?

Reply:
- Disabled: app links directly to landing page
- Enabled: must enter event code to access landing page/website
The above is the description provided in the user story, I assumed that I achieved the functionality by redirecting the user to index.html bypassing the login page if event code is disabled.
I am not sure whether it's the right way to do it. Please let me know the right way to do it, if I am wrong here.